Update eProsima Shapes Demo to v2.0.0 [8679] (#8)
eProsima/Shapes-Demo-Docs: docs/examples/deadline.rst

.. _examples_deadline:

Deadline
========

The Deadline QoS raises an alarm when the frequency of new samples falls below a predefined threshold.
This policy can be useful for those cases which need periodical updates.

> It can be distinguished between publisher and subscriber deadline period.

Reviewer: Who is "it"? Something like there exists a distinction between... would sound better

> On the publisher side, that period defines the maximum interval between writes, while the subscriber establishes

Reviewer: ```suggestion
On the publisher side, that period defines the maximum interval between writes, while on the subscriber's, it establishes
```

> the maximum interval in which the reader should receive a new sample.

Reviewer: ```suggestion
the maximum interval in which the reader expects to receive a new sample.
```

In this test, a publisher and a subscriber with a deadline period higher than the write rate
of the publisher will be created.
That shows the normal behavior of the system.
Then the write rate will be increased to illustrate the effects of non-compliance with the deadline.

Step-by-step example implementation
-----------------------------------

First, launch two instances and create a publisher and a subscriber:

1. Create a red square publisher:

   - Start eProsima Shapes Demo (this instance will be referred to as *Instance1*).
   - Click on Publish.
   - Select SQUARE option for Shape.
   - Select RED for Color.
   - Set Deadline Duration to 100. (The default write rate is 75 ms)

2. Create a square subscriber:

   - Start eProsima Shapes Demo (this instance will be referred to as *Instance2*).
   - Click on Subscribe.
   - Select SQUARE option for Shape.
   - Set a value for the Deadline Duration higher or equal to the one stated for the publisher.

   .. warning::

      If the value of the subscriber Deadline Duration is lower than the value stated for the publisher
      the entities will not match.

> It can be seen now the normal behavior of the system, where the publisher is sending new samples and the subscriber

Reviewer: ```suggestion
In the following figure, the normal behavior of the system can be seen, where the publisher is sending new samples and the subscriber
```
